ctcp/ctcp.go

```diff
@@ -0,0 +1,30 @@
+package ctcp
+
+import "errors"
+
+// Message contains text, the text's type, and the next part of the message, if there is one
+type Message struct {
+	value       string
+	textType    TextType
+	nextMessage *Message
```

Comment:
What is the usage of `nextMessage`?

Reply:
> If the user actor wants to query the USERINFO of user victim, and is
> in the middle of a conversation, the client may decide to tack on
> USERINFO request on the end of a normal text message. Let's say actor
> wants to send the textmessage "Say hi to Ron\n\t/actor" and the CTCP
> request "USERINFO" to victim:
> 
> 	PRIVMSG victim :Say hi to Ron\n\t/actor
> 
> plus:
> 
> 	USERINFO

This case can be achieved by allowing messages to be chained together.